cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/KafkaConnectS2ICluster.java

```diff
@@ -0,0 +1,438 @@
+                .endMetadata()
+                .withNewSpec()
+                    .withLookupPolicy(new ImageLookupPolicyBuilder().withLocal(false).build())
+                    .withTags(sourceTag)
+                .endSpec()
+                .build();
+
+        return imageStream;
+    }
+
+    /**
+     * Generate new target ImageStream
+     *
+     * @return      Target ImageStream resource definition
+     */
+    public ImageStream generateTargetImageStream() {
+        ImageStream imageStream = new ImageStreamBuilder()
+                .withNewMetadata()
+                    .withName(name)
+                    .withNamespace(namespace)
+                    .withLabels(getLabelsWithName())
+                .endMetadata()
+                .withNewSpec()
+                    .withLookupPolicy(new ImageLookupPolicyBuilder().withLocal(true).build())
+                .endSpec()
+                .build();
+
+        return imageStream;
+    }
+
+    /**
+     * Generate new BuildConfig
+     *
+     * @return      BuildConfig resource definition
+     */
+    public BuildConfig generateBuildConfig() {
+        BuildTriggerPolicy triggerConfigChange = new BuildTriggerPolicy();
+        triggerConfigChange.setType("ConfigChange");
+
+        BuildTriggerPolicy triggerImageChange = new BuildTriggerPolicy();
+        triggerImageChange.setType("ImageChange");
+        triggerImageChange.setImageChange(new ImageChangeTrigger());
+
+        BuildConfig build = new BuildConfigBuilder()
+                .withNewMetadata()
+                    .withName(name)
+                    .withLabels(getLabelsWithName())
+                    .withNamespace(namespace)
+                .endMetadata()
+                .withNewSpec()
+                    .withFailedBuildsHistoryLimit(5)
+                    .withNewOutput()
+                        .withNewTo()
+                            .withKind("ImageStreamTag")
+                            .withName(image)
+                        .endTo()
+                    .endOutput()
+                    .withRunPolicy("Serial")
+                    .withNewSource()
+                        .withType("Binary")
+                        .withBinary(new BinaryBuildSource())
+                    .endSource()
+                    .withNewStrategy()
+                        .withType("Source")
+                        .withNewSourceStrategy()
+                            .withNewFrom()
+                                .withKind("ImageStreamTag")
+                                .withName(getSourceImageStreamName() + ":" + sourceImageTag)
+                            .endFrom()
+                        .endSourceStrategy()
+                    .endStrategy()
+                    .withTriggers(triggerConfigChange, triggerImageChange)
+                .endSpec()
+                .build();
+
+        return build;
+    }
+
+    /**
+     * Patches DeploymentConfig
```

Comment:
I guess now the patch operations mutate the passed resource, (rather than getting the resource for themselves) we should either copy the resource, or at least document that it mutates the given resource.

Reply:
Yeah ... I was thinking about that. In the go client you normally do deep copy of the object before modifying it. Because they are actually stored in some shared cache. So by modifying them you modify the cache. But I'm not sure how that works with the Fabric8 API. Anyway, I would say that this is again for a separate issue / PR, because this isn't anything specific for the S2I feature.